org.civicrm.styleguide/CRM/Styleguide/Manager.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+class CRM_Styleguide_Manager {
+
+  /**
+   * @var array
+   */
+  private $all = NULL;
+
+  /**
+   * Get the definition of specific style-guide by name.
+   *
+   * @param string $name
+   * @return array|NULL
+   *   If the style-guide exists, it is an array with properties:
+   *     - name: string
+   *     - label: string
+   */
+  public function get($name) {
+    $all = $this->getAll();
+    return @$all[$name];
+  }
+
+  /**
+   * Get a list of all style guides.
+   *
+   * @return array
+   */
+  public function getAll() {
+    if ($this->all === NULL) {
+      $extPath = CRM_Core_Resources::singleton()->getPath('org.civicrm.styleguide');
+
+      $this->all = array(
+        'crmstar' => array(
+          'label' => ts('crm-*'),
+          'path' => "{$extPath}/guides/crmstar",
+        ),
+        'bootstrap' => array(
+          'label' => ts('Bootstrap'),
+          // FIXME: 'path' => "{$extPath}/guides/bootstrap",
+          'path' => "{$extPath}/partials",
+        ),
+        'bootstrapcivicrm' => array(
+          'label' => ts('Bootstrap-CiviCRM'),
+          'path' => "{$extPath}/guides/bootstrapcivicrm",
+        ),
+        // FIXME: Consider moving declaration to another extension.
+        'bootstrapcivihr' => array(
+          'label' => ts('Bootstrap-CiviHR'),
+          'path' => "{$extPath}/guides/bootstrapcivihr",
+        ),
+      );
+
+      CRM_Utils_Hook::singleton()->invoke(1, $this->all,
```

Comment:
I think it would be better to not expose the property like this (it's is private after all, isn't?). What do you think of this?:
- Add a public `addStyleGuide()` method to CRM_Styleguide_Manager
- Pass the CRM_Styleguide_Manager instance to the hooks, so people can use the `addStyleGuide()` instead of inserting everything in to the array

This will also avoid people from removing style guides from array. If that's is not desired, then you can all add a `removeStyleGuide()` method.